### PR TITLE
Initial implementation for gperf port

### PR DESCRIPTION
**Pull request by matwey**
Initial implementation for gperf port

- What does your PR fix? Fixes issue #9243

Add gperf port

- Which triplets are supported/not supported? Have you updated the CI baseline?
linux-x64 and windows have been tested.

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?

### CODE REVIEW

File: ports/gperf/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,31 @@
+include(vcpkg_common_functions)
```
Comment (JackBoosY) [inline ports/gperf/portfile.cmake]: We don't need to include this, please remove this line.

File: ports/gperf/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+file(COPY ${CMAKE_CURRENT_LIST_DIR}/config.h.in DESTINATION ${SOURCE_PATH})
+
+vcpkg_configure_cmake(
+    SOURCE_PATH ${SOURCE_PATH}
+    OPTIONS_RELEASE -DCMAKE_INSTALL_BINDIR=tools
+)
+
+vcpkg_install_cmake()
+
+file(COPY ${SOURCE_PATH}/COPYING DESTINATION ${CURRENT_PACKAGES_DIR}/share/gperf)
```
Comment (JackBoosY) [inline ports/gperf/portfile.cmake]: Please replace these 2 line with:
`file(INSTALL ${SOURCE_PATH}/COPYING DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)`

File: ports/gperf/portfile.cmake [hunk 3]
```diff
@@ -0,0 +1,31 @@
+include(vcpkg_common_functions)
+
```
Comment (JackBoosY) [inline ports/gperf/portfile.cmake]: Seems gperf cannot build in uwp, please add `vcpkg_fail_port_install(ON_TARGET "UWP")` here. and add the following code to _VCPKG_PATH/scripts/ci.baseline.txt_ :
```
gperf:x64-uwp=fail
gperf:arm-uwp=fail
```

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: ports/gperf/portfile.cmake (hunk 1 above)
Comment (matwey) [reply ports/gperf/portfile.cmake]: Done

File: ports/gperf/portfile.cmake (hunk 2 above)
Comment (matwey) [reply ports/gperf/portfile.cmake]: Done

File: ports/gperf/portfile.cmake (hunk 3 above)
Comment (matwey) [reply ports/gperf/portfile.cmake]: Done

**Review by JackBoosY — APPROVED** [review]

**Review by PhoebeHui — APPROVED** [review]

**Comment by ubsan** [comment]
Merged manually due to whitespace changes

**Pull request closed without merging**